### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by diegosalvi** [apache/pulsar#19678]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

Pulsar 2.9.2 running on Linux

### Minimal reproduce step
Unfortunately I don't have a reproducer

On a production environment I create a new reader from a shared PulsarClient for a specific topic and starting from a specific message id then I ask if it has any message available.
On production environment happened some time that the PulsarReader#hasMessageAvailable throw an exception and then the thread remains stuck forever on Reader#close (Autocloseable).

```
try (Reader<byte[]> pulsarReader = pulsarClient.newReader()
                .startMessageId(lastMessageId)
                .topic(topicName)
                .create()) {

        if (pulsarReader.hasMessageAvailable()) {
            //business logic
        }
}
```

I didn't see any strange log but one (I don't know if can be related).

org.apache.pulsar.client.impl.ConsumerImpl-23-02-25.log.gz
```
23-02-25-20-59-46	[xxx/ns/topic][reader-3552097eea] Subscribing to topic on cnx [id: 0x5b9863f4, L:/10.10.10.10:38350 - R:broker1/10.10.20.10:6650], consumerId 8950338
23-02-25-20-59-46	[xxx/ns/topic][reader-3552097eea] Subscribed to topic on broker1/10.10.20.10:6650 -- consumer: 8950338
23-02-25-20-59-46	[xxx/ns/topic][reader-3552097eea] Get topic last message Id
23-02-25-20-59-46	[xxx/ns/topic][reader-3552097eea] Failed getLastMessageId command
```

org.apache.pulsar.client.impl.ClientCnx-23-02-25.log.gz
Here for the same id (cxt?) i see an error for a different topic on a differen tenant
```
23-02-25-20-59-46       [id: 0x5b9863f4, L:/10.10.10.10:38350 - R:broker1/10.10.20.10:6650] Close connection because received internal-server error java.lang.IllegalStateException: Namespace bundle yyyy/ns/0x00000000_0x40000000 is being unloaded
23-02-25-20-59-46       [id: 0x5b9863f4, L:/10.10.10.10:38350 ! R:broker1/10.10.20.10:6650] Failed to send GetLastMessageId request to broker: null
23-02-25-20-59-46       [id: 0x5b9863f4, L:/10.10.10.10:38350 ! R:broker1/10.10.20.10:6650] Disconnected
```

org.apache.pulsar.client.impl.ConnectionHandler-23-02-25.log.gz
Here I see the "Ignoring reconnection request" log for the stuck reader
```
23-02-25-20-59-46       [yyyy/ns/other] [reader-dd22ec2f08] Error connecting to broker: org.apache.pulsar.client.api.PulsarClientException$LookupException: {"errorMsg":"java.lang.IllegalStateException: Namespace bundle yyyy/
ns/0x00000000_0x40000000 is being unloaded","reqId":3420261376469640774, "remote":"broker1/10.10.20.10:6650", "local":"/10.10.10.10:38350"}
23-02-25-20-59-46       [yyyy/ns/other] [reader-dd22ec2f08] Could not get connection to broker: org.apache.pulsar.client.api.PulsarClientException$LookupException: {"errorMsg":"java.lang.IllegalStateException: Namespace bundle 
idc6966/ns/0x00000000_0x40000000 is being unloaded","reqId":3420261376469640774, "remote":"broker1/10.10.20.10:6650", "local":"/10.10.10.10:38350"} -- Will try again in 0.1 s
23-02-25-20-59-46       [xxx/ns/topic] [reader-3552097eea] Ignoring reconnection request (state: Closing)
```

### What did you expect to see?

The PulsarReader should be closed

### What did you see instead?

Current thread was stuck on PulsarReader#close waiting for async Consumer close to be completed.
I had to kill the JVM.
```
"statengineXXXX" #93458 prio=5 os_prio=0 cpu=11730.69ms elapsed=2129.57s tid=0x00007f86407fa9b0 nid=0x7e31f waiting on condition  [0x00007f854e2f9000]
   java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.3/Native Method)
        - parking to wait for  <0x00000014197b9e08> (a java.util.concurrent.CompletableFuture$Signaller)
        at java.util.concurrent.locks.LockSupport.park(java.base@17.0.3/LockSupport.java:211)
        at java.util.concurrent.CompletableFuture$Signaller.block(java.base@17.0.3/CompletableFuture.java:1864)
        at java.util.concurrent.ForkJoinPool.unmanagedBlock(java.base@17.0.3/ForkJoinPool.java:3463)
        at java.util.concurrent.ForkJoinPool.managedBlock(java.base@17.0.3/ForkJoinPool.java:3434)
        at java.util.concurrent.CompletableFuture.waitingGet(java.base@17.0.3/CompletableFuture.java:1898)
        at java.util.concurrent.CompletableFuture.get(java.base@17.0.3/CompletableFuture.java:2072)
        at org.apache.pulsar.client.impl.ConsumerBase.close(ConsumerBase.java:596)
        at org.apache.pulsar.client.impl.ReaderImpl.close(ReaderImpl.java:171)
[...]

   Locked ownable synchronizers:
        - <0x00000010bf00d948> (a java.util.concurrent.ThreadPoolExecutor$Worker)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by lhotari** [comment]
@diegosalvi Please add a title for this issue

**Comment by eolivelli** [comment]
`Namespace bundle yyyy/ns/0x00000000_0x40000000 is being unloaded`

This happens when a bundle is being unloaded, typically this happens when you shutdown the broker or when the LoadBalancer decides to move the bundle (set of topics) to another broker in order to shed the load.
This is usually not a problem and the system should recover (the reader should always retry).



There are a few bug fixes about "consumers" and "readers" in 2.9.3.
From the description I cannot point to any specific fix
Having an heap dump of the problem will allow to understand what is the Reading waiting for.


https://github.com/apache/pulsar/pulls?q=is%3Apr+label%3Arelease%2F2.9.3+consumer

I would suggest to try to upgrade to 2.9.3 or to 2.10.3.

**Comment by diegosalvi** [comment]
@lhotari sorry but I don't have the permission to change the issue name :(
I'll upgrade to latest 2.10.x and see if it happens again
